riscv/processor.h

```diff
@@ -232,12 +232,17 @@ class processor_t : public abstract_device_t
     return extension_enabled_const(isa_extension_t(ext));
   }
   bool extension_enabled_const(isa_extension_t ext) const {
-    if (ext >= 'A' && ext <= 'Z')
+    if (ext >= 'A' && ext <= 'Z') {
       return state.misa->extension_enabled_const(ext);
-    else
-      return isa->extension_enabled(ext);  // assume this can't change
+    } else {
+      assert(!extension_dynamic[ext]);
+      extension_assumed_const[ext] = true;
+      return extension_enabled(ext);
```

Comment:
This way seems not necessary now since extension_enabled_const(EXT_ZCMT) has been changed to  extension_enabled(EXT_ZCMT) in the previous commit and we can use dynamic extensions directly here and  which initialize it just from isa string.

Reply:
Right.  But I think it was @scottj97's intention to future-proof the implementation--which, given he was able to do without duplicating the complicated enable logic, makes sense to me.